addons/cert-manager/addon.rb

```diff
@@ -56,45 +58,34 @@ def self.messages
     stack = kube_stack
 
     if config.ca_issuer&.enabled
+      logger.info "Enabling kubernetes CA issuer ..."
       stack.resources << build_ca_secret
-      stack.resources << build_ca_issuer
+      config.issuers << build_ca_issuer.to_h
     end
 
-    stack.apply(kube_client)
-
-    migrate_le_acme_issuers
-    migrate_le_acme_cluster_issuers
-  }
-
-  def patch_spec
-    {
-      acme: {
-        server: 'https://acme-v02.api.letsencrypt.org/directory'
-      }
-    }
-  end
-
-  def le_acme_v1_endpoint
-    'https://acme-v01.api.letsencrypt.org/directory'
-  end
-
-  def migrate_le_acme_issuers
-    kube_client.api('certmanager.k8s.io/v1alpha1').resource('issuers', namespace: nil).list.each do |issuer|
-      next unless issuer&.spec&.acme&.server == le_acme_v1_endpoint
-
-      rc = kube_client.client_for_resource(issuer, namespace: issuer.metadata.namespace)
-      rc.merge_patch(issuer.metadata.name, { spec: patch_spec }, namespace: issuer.metadata.namespace, strategic_merge: false)
+    if config.issuer
+      logger.warn "Issuer config option is deprecated, use issuers instead."
+      config.issuers << build_legacy_issuer.to_h
     end
-  end
 
-  def migrate_le_acme_cluster_issuers
-    kube_client.api('certmanager.k8s.io/v1alpha1').resource('clusterissuers', namespace: nil).list.each do |issuer|
-      next unless issuer&.spec&.acme&.server == le_acme_v1_endpoint
+    stack.apply(kube_client)
 
-      rc = kube_client.client_for_resource(issuer)
-      rc.merge_patch(issuer.metadata.name, { spec: patch_spec }, strategic_merge: false)
+    unless config.issuers.empty?
+      logger.info "Applying issuers (this might take a moment) ..."
+      issuers = []
+      config.issuers.each do |i|
+        issuers << K8s::Resource.new(
+          i.merge(
+            apiVersion: "certmanager.k8s.io/v1alpha1"
+          )
+        )
+      end
+      issuers_stack = Pharos::Kube::Stack.new('cert-manager-issuers', issuers)
+      Retry.perform(300, exceptions: [K8s::Error::InternalError]) do
+        issuers_stack.apply(kube_client)
```

Comment:
I'd guess we also need to handle the `cert-manager-issuers` stack separately whendisabling the addon?

Reply:
Good point. Although I'm not sure what happens to CRD resources if CRD is removed? Do they get garbage collected automatically?

Reply:
They should get GC'd:
> When you delete a CustomResourceDefinition, the server will uninstall the RESTful API endpoint and delete all custom objects stored in it.

> If you later recreate the same CustomResourceDefinition, it will start out empty.